File: LoanManagementApp02/LoanManagementApp02/LoanManagementApp02/Models/Loan.js
// Defines the Loan model in MongoDB.
const mongoose = require('mongoose');

const LoanSchema = new mongoose.Schema({
    user_id: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true }, // User reference
    amount: { type: Number, required: true }, // Loan amount
    status: { type: String, enum: ['Pending', 'Approved', 'Rejected', 'Disbursed'], default: 'Pending' },
    created_at: { type: Date, default: Date.now } // Loan creation timestamp
});

module.exports = mongoose.model('Loan', LoanSchema);

File: LoanManagementApp02/LoanManagementApp02/LoanManagementApp02/Models/Loan.cs
using LoanManagementSystem.Models;
using LoanManagementSystem.Models.Enums;

namespace LoanManagementSystem.Models
{
    public class Loan
    {
        public Guid Id { get; set; }
        public Guid UserId { get; set; }
        public User User { get; set; }
        public decimal Amount { get; set; }
        public LoanStatus Status { get; set; } // Pending, Approved, Rejected, Disbursed
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}
